packages/core-data/src/entity-provider.js

```diff
@@ -0,0 +1,71 @@
+/**
+ * WordPress dependencies
+ */
+import { createContext, useContext, useCallback } from '@wordpress/element';
+import { useSelect, useDispatch } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import { defaultEntities } from './entities';
+
+const entities = {
+	...defaultEntities.reduce( ( acc, entity ) => {
+		acc[ entity.name ] = { kind: entity.kind, context: createContext() };
+		return acc;
+	}, {} ),
+	post: { kind: 'postType', context: createContext() },
```

Comment:
If I remember properly, there's no way to ensure the name is unique across kinds. so we might be better using a two level objects [kind][name] or just an array of entities.

Also, there might be a better way to automatically create contexts for these dynamic entities (like post).

Reply:
>be better using a two level objects [kind][name]

Yes

>Also, there might be a better way to automatically create contexts for these dynamic entities (like post).

We could add this to the store and load them asynchronously, but it seems unnecessarily complex.

Reply:
Or.....

https://github.com/WordPress/gutenberg/pull/17153/commits/217bcad354421952a0508342d46c2ba1fc5fcdd1

:rocket::rocket::rocket::rocket::rocket::rocket::rocket::rocket::rocket::laughing: